CMakeLists.txt

```diff
@@ -15,54 +15,89 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-find_package(GDAL 3 REQUIRED)
 find_package(ament_cmake REQUIRED)
+find_package(grid_map_msgs REQUIRED)
 find_package(grid_map_ros REQUIRED)
 find_package(grid_map_core REQUIRED)
-find_package(Eigen3 REQUIRED)
+find_package(rclcpp REQUIRED)
 
-add_library(${PROJECT_NAME})
-add_subdirectory(src)
+find_package(Eigen3 REQUIRED)
+find_package(GDAL 3 REQUIRED)
 
-target_include_directories(${PROJECT_NAME}
-    PUBLIC
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
-)
+# macOS
+if(APPLE)
+  link_directories(/opt/homebrew/opt/yaml-cpp/lib)
+endif()
 
 # Reverse compatability for GDAL<3.5
 # https://gdal.org/development/cmake.html
 if(NOT TARGET GDAL::GDAL)
-    add_library(GDAL::GDAL ALIAS ${GDAL_LIBRARY})
+  add_library(GDAL::GDAL ALIAS ${GDAL_LIBRARY})
 endif()
 
-target_link_libraries(${PROJECT_NAME} Eigen3::Eigen GDAL::GDAL grid_map_core::grid_map_core)
+# Build
+add_library(${PROJECT_NAME}
+  src/grid_map_geo.cpp
+)
+
+target_include_directories(${PROJECT_NAME}
+  PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+  ${GDAL_INCLUDE_DIR}
+)
+
+target_link_libraries(${PROJECT_NAME} Eigen3::Eigen GDAL::GDAL)
+
+ament_target_dependencies(${PROJECT_NAME} SYSTEM
+  grid_map_core
```

Comment:
grid_map_core was already linked before through the namespaced target (preferred) compared to `ament_target_dependencies`. In conversations with the ROS maintainers, seems like the only thing `ament_target_dependencies` exists for these days is message packages because those can't use namespaced export targets yet.

Reply:
If:

```cmake
ament_target_dependencies(${PROJECT_NAME} SYSTEM
  grid_map_core
  grid_map_ros
)
```

is replaced with:

```bash
target_link_libraries(${PROJECT_NAME}
  Eigen3::Eigen
  GDAL::GDAL
  grid_map_core
  grid_map_ros
)
```

then there is an error:

```bash
In file included from /Users/rhys/Code/ros2/humble/ros2-aerial/src/grid_map_geo/src/grid_map_geo.cpp:40:
In file included from /Users/rhys/Code/ros2/humble/ros2-aerial/src/grid_map_geo/include/grid_map_geo/grid_map_geo.hpp:37:
In file included from /Users/rhys/Code/ros2/humble/ros2-aerial/src/grid_map_geo/include/grid_map_geo/transform.hpp:49:
In file included from /opt/homebrew/include/eigen3/Eigen/Dense:1:
In file included from /opt/homebrew/include/eigen3/Eigen/Core:256:
/opt/homebrew/include/eigen3/Eigen/src/Core/functors/StlFunctors.h:159:10: fatal error: 'grid_map_core/eigen_plugins/FunctorsPlugin.hpp' file not found
#include EIGEN_FUNCTORS_PLUGIN
         ^~~~~~~~~~~~~~~~~~~~~
<command line>:2:31: note: expanded from macro 'EIGEN_FUNCTORS_PLUGIN'
#define EIGEN_FUNCTORS_PLUGIN "grid_map_core/eigen_plugins/FunctorsPlugin.hpp"
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

The target `grid_map_core::grid_map_core` is not available. Something needing fixing upstream?

Reply:
Ah yes, try my patch here: https://github.com/ANYbotics/grid_map/pull/404

Since `grid_map` seems largely unmaintained, we may have trouble getting changes in.